src/tbm/NFT.scilla

```diff
@@ -133,9 +163,18 @@ field token_uris: Map Uint256 String = Emp Uint256 String
 (* Total supply of tokens *)
 field total_supply: Uint256 = Uint256 0
 
+(* Tokens reserved to be minted by GiveawayMinter *)
+field total_reserved: Uint256 = Uint256 0
+
 (* Base URI that will be updated once upon reveal *)
 field base_uri: String = "https://d5gozuhfenzxt.cloudfront.net/metadata/"
 
+(* Offset index after sale ends *)
+field offset_index: Uint256 = Uint256 0
+
+(* One-time flag to ensure offset applied only once *)
+field is_offset_applied: Bool = False
```

Comment:
not necessary, use `offset_index > 0` as check for whether value is set.

Reply:
@tanzhuoyao-nus use `Option {Uint256}` if need be